lib/Service/DocumentService.php

```diff
@@ -152,17 +152,45 @@ public function createDocument(File $file): Document {
 	}
 
 	/**
-	 * @param $document
+	 * @param $documentId
 	 * @return ISimpleFile
 	 * @throws NotFoundException
 	 */
-	public function getStateFile($document): ISimpleFile {
+	public function getStateFile($documentId): ISimpleFile {
+		$filename = (string)$documentId . '.yjs';
 		if (!$this->ensureDocumentsFolder()) {
 			throw new NotFoundException('No app data folder present for text documents');
 		}
-		return $this->appData->getFolder('documents')->getFile((string) $document);
+		return $this->appData->getFolder('documents')->getFile($filename);
+	}
+
+	/**
+	 * @param $documentId
+	 * @return ISimpleFile
+	 * @throws NotPermittedException
+	 */
+	public function createStateFile($documentId): ISimpleFile {
+		$filename = (string)$documentId . '.yjs';
+		return $this->appData->getFolder('documents')->newFile($filename);
+	}
+
+	/**
+	 * @param $documentId
+	 * @param $content
+	 */
+	public function writeDocumentState($documentId, $content): void {
+		try {
+			$documentStateFile = $this->getStateFile($documentId);
+		} catch (NotFoundException $e) {
+			$documentStateFile = $this->createStateFile($documentId);
+		} catch (NotPermittedException $e) {
+			$this->logger->error('Failed to create document state file', ['exception' => $e]);
+			return;
```

Comment:
Is it intended that `writeDocumentState()` doesn't fail in case of `NotPermittedException`? I'm just unsure here.

Reply:
Yes... it's an optimization - if we cannot write the state file it's most likely because we cannot create the documents folder or so.

That's a problem with the setup that will prevent using the document state to load an editing session - but text can recover from that. So no use failing hard here I think.